aten/src/ATen/native/TensorTransformations.cpp

```diff
@@ -57,6 +57,37 @@ Tensor flip_cpu(const Tensor& self, IntList dims) {
   return out_tensor;
 }
 
+Tensor roll_cpu(const Tensor& self, IntList shifts, IntList dims) {
+  // todo: support rolling along no or multiple dimensions as in numpy.roll.
+  AT_CHECK(dims.size() == 1, "only single dimension roll currently supported");
+  AT_CHECK(shifts.size() == dims.size(), "shifts and dimensions must align");
+  // If the first dimension is zero, this is an empty tensor and rolls do nothing.
+  // Return a clone so the caller can safely modify result, and avoid a div by
+  // zero error below.
+  if( self.size(0) == 0 ) {
+    return self.clone();
+  }
+  int64_t dim = dims[0];
+  int64_t size = self.size(dim);
+  int64_t start = (size - shifts[0]) % size;
+  // Behavior of % is different in C++ vs Python for negative numbers. This
+  // corrects the difference.
+  if( start < 0 ) start = start + size;
+
+  std::vector<Tensor> vec = std::vector<Tensor>(size);
+  int64_t index = 0;
+  for (int64_t i = start; i < size; i++) {
+    vec[index++] = self.select(dim, i);
```

Comment:
Backward of this would be really inefficient because each `select` creates a full grad_input tensor! Could you change to using `unbind` instead?

Reply:
updated to use unbind